cogs/stats/leaderboard.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+from typing import NamedTuple
+from datetime import datetime
+from datetime import timedelta as td
+
+import asyncpg
+import discord
+from discord.ext import commands
+
+from utils import HideoutCog, HideoutContext
+from bot import HideoutManager
+
+
+class DatabaseData(NamedTuple):
+    rank: int
+    message_count: int
+
+
+class LeaderboardCard:
+    # Options
+    WIDTH = 1200
+    HEIGHT = 2400
+    OVERALL_PADDING = 50
+    LEFT_TEXT_PADDING = 20
+
+    DROP_SHADOW_OFFSET = (3, 3)
+    DROP_SHADOW_ITERATIONS = 15
+    DROP_SHADOW_EXTRA_SIZE = 10
+
+    BG_COLOR = discord.Color.from_str("#1b1d21")
+
+    """WIP. Using embed for now"""
+
+
+class LeaderboardView(discord.ui.View):
+    def __init__(self, embed: LeaderboardEmbed, author: discord.User | discord.Member):
+        self.author = author
+        self.current_embed: LeaderboardEmbed = embed
+        super().__init__(timeout=300)
```

Comment:
Here you should subclass `utils.View` instead, so errors properly log to the webhook, and take a parameter of `bot` which gets passed to super init.